### ISSUE TITLE
Layout/MultilineMethodCallIndentation still glitches with multiple blocks

### ISSUE THREAD

**Issue by zverok** [rubocop/rubocop#15526]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Layout/MultilineMethodCallIndentation

### Steps to reproduce the problem

test.rb:
```ruby
# frozen_string_literal: true

users
  .group_by { _1[:name] }.transform_values { _1.pluck(:id) }
  .each do |name, user_ids|
  p(name:, user_ids:)
end
```

This is similar to the example in #15122 (which was fixed).

### Expected behavior

The layout should be considered correct (at least in regards to `.each` indentation, the block insides are debatable, but not the scope of the cop).

### Actual behavior

```
test.rb:5:3: C: [Correctable] Layout/MultilineMethodCallIndentation: Align .each with .transform_values on line 4.
  .each do |name, user_ids|
  ^^^^^
```

Autocorrection:
```ruby
users
  .group_by { _1[:name] }.transform_values { _1.pluck(:id) }
                         .each do |name, user_ids|
  p(name:, user_ids:)
end
```

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
$ rubocop -V
1.89.0 (using Parser 3.3.12.0, Prism 1.9.0, rubocop-ast 1.50.0, analyzing as Ruby 4.0, running on ruby 4.0.3) [x86_64-linux]
```

**Comment by zverok** [comment]
cc @hammadxcm 🙏
